python/cog/server/runner.py

```diff
@@ -88,8 +93,21 @@ def predict(
 
         task_kwargs = task_kwargs or {}
 
-        self._predict_task = PredictTask(prediction, **task_kwargs)
-        self._prediction_id = prediction.id
+        tag = prediction.id
+        if tag is None:
+            tag = uuid.uuid4().hex
```

Comment:
The external interface is that callers can cancel predictions by providing the prediction id. If we were to always use uuids for tags, we'd need to maintain a mapping from prediction id to tag. I'd rather not do that.

Reply:
So should it be an error if we don't have a `prediction.id`?

Reply:
No, it's currently allowed to not provide a prediction id, it just means you can't then cancel predictions. 

We could make it required, that'd be a breaking change but I don't imagine it would actually break many users.

Reply:
> Final thought: should we be setting prediction.id if it's not already set here?

I do think it's worth considering whether auto generating a prediction id is a useful thing, it likely is. However as this would change the existing behavior I think it's best left out of this PR.

Reply:
actually i think this is important, otherwise we will default to `None` when we record in `self._predict_tasks` and we will overwrite any running task. This will make the busy check report us as not busy even if we're actually at max concurrency.

I'll push a fix to do this

Reply:
I discussed with @aron and I'm wrong and he's right. (I briefly pushed a commit that always set prediction.id but that wasn't needed and we're backing it out.)

Setting prediction.id is weird because it changes the task we give back to the user.

We don't need to set it (contrary to my comment above) because we key _predict_tasks on `tag`, not on `prediction.id`.